src/main/java/chess/model/piece/Queen.java

```diff
@@ -0,0 +1,35 @@
+package chess.model.piece;
+
+import chess.model.position.ChessPosition;
+import chess.model.position.Distance;
+
+import java.util.List;
+
+public class Queen extends Piece {
+
+    public Queen(Side side) {
+        super(side);
+    }
+
+    @Override
+    public String getText() {
+        if (side.isWhite()) {
+            return "q";
+        }
+        return "Q";
+    }
+
+    @Override
+    public List<ChessPosition> findPath(ChessPosition source, ChessPosition target, Piece targetPiece) {
+        checkValidTargetPiece(targetPiece);
+        Distance distance = target.calculateDistance(source);
+        if (canMove(distance)) {
+            return distance.findPath(source);
+        }
+        return List.of();
+    }
+
+    private boolean canMove(Distance distance) {
+        return distance.isCrossMovement() || distance.isDiagonalMovement();
```

Comment:
```suggestion
        if (distance.isCrossMovement()) {
            return true;
        }
        return distance.isDiagonalMovement();
```
(반영하지 않으셔도 됩니다.) 개인적으로 &&와 || 연산자는 가독성을 해친다라고 생각하기 때문에 if문으로 좀 더 직관적으로 표현하는게 좋다라고 생각합니다 :)

Reply:
저도 반복된 &&와 || 때문에 헷갈리는 부분이 많았어서 3개 이상부터는 아래와 같이 리팩토링하였습니다..! 추천 감사합니다 👍

```java
    private boolean canMove(Distance distance) {
        if (distance.isCrossMovement() || distance.isDiagonalMovement()) {
            return false;
        }
        return distance.hasSame(DISPLACEMENT);
    }
```